src/cmap/connection.ts

```diff
@@ -509,14 +513,29 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
             new CommandSucceededEvent(
               this,
               message,
-              options.noResponse ? undefined : document,
+              options.noResponse ? undefined : document.toObject(options),
```

Comment:
This and the other instances of toObject make the deep clone we preform for command monitoring redundant. Should we remove that here or should I strive to pass through the same instance? @baileympearson (sorry should've called this out sooner)

Reply:
Should we be caching the full deserialized object in `MongoDBResponse.toObject()`?  If so, wouldn't we still need the deep clone?

It seems like caching would be necessary to avoid redundant deserialization when command monitoring is enabled

Reply:
Yea, we can defer changing this, I'll reuse the same deserialized object and leave the deep cloning be. 

I say defer because I think at one point we considered that deserializing twice would be better than cloning, we can make that choice intentionally later.